dotcom-rendering/src/web/lib/parser/jsonParser.ts

```diff
@@ -0,0 +1,16 @@
+export const safeJsonParse =
+	<T>(guard: (o: unknown) => o is T) =>
+	(text: string): ParseResult<T> => {
+		try {
+			const parsed = JSON.parse(text) as unknown;
+			return guard(parsed)
+				? { parsed, hasError: false }
+				: { hasError: true };
+		} catch (_) {
+			return { hasError: true };
+		}
+	};
+
+type ParseResult<T> =
+	| { parsed: T; hasError: false; error?: undefined }
+	| { parsed?: undefined; hasError: true; error?: unknown };
```

Comment:
I notice I don't think we ever use the `error` parameter here? perhaps it can be removed from the type definitions?

Reply:
yeah, that's a good point. Removed it